frontend/src/services/api/ApiService.ts

```diff
@@ -0,0 +1,65 @@
+import type { Endpoint, EndpointReq, EndpointRes } from "./endpoints/endpoint";
+
+const BASE_URL = import.meta.env.VITE_BACKEND_URL;
+
+class ApiService {
+    /**
+     * Perform a fetch to the provided endpoint, with one re-attempt if initial call fails.
+     * 
+     * @param endpoint 
+     * @param payload 
+     * @param retry 
+     * @returns 
+     */
+    public async call<E extends Endpoint<any, any>>(
+        endpoint: E,
+        payload?: EndpointReq<E>,
+        accessToken?: string | null,
+    ): Promise<EndpointRes<E>> {
+        
+        const url = new URL(`api${endpoint.route}`, BASE_URL);
+        const request = this.buildRequest(endpoint, accessToken, payload);
+
+        let response = await fetch(url.toString(), request);
+
+        if (!response.ok) {
+            throw new Error(`Request failed with status ${response.status}`);
+        }
+
+        const parsedResponse = await response.json() as EndpointRes<E>;
+
+        return parsedResponse;
+    }
+
+    /**
+     * Build the request with headers, payload, method, etc.
+     * 
+     * @param endpoint 
+     * @param payload 
```

Comment:
```suggestion
     * @param endpoint 
     * @param accessToken
     * @param payload 
```

Missed the following param here. Alternatively, we could just not have these docblocks altogether. I mentioned this in another comment but these docblocks on their own without comments are pretty useless and just becomes another thing to maintain as seen in the missed docs. It would probably be better to just add these when we have a meaningful comment to go along with it.

Reply:
You're right, they seem to be just bloating the files as well. I'll just keep the main comment instead of adding the `@param` and `@return`.